core/src/main/kotlin/xtdb/trie/HashTrie.kt

```diff
@@ -67,6 +86,14 @@ interface HashTrie<N : Node<N>, L : N> {
     }
 }
 
+abstract class BaseHashTrie<N : Node<N>, L : N>(
```

Comment:
this probably goes if you have hash-bucketing as its own class - 'composition rather than inheritance', as they say